pinc/filter_project_list.inc

```diff
@@ -11,485 +12,418 @@ include_once($relPath.'misc.inc'); // attr_safe()
-    $result = mysqli_query(DPDatabase::get_connection(), $query);
-    $row = mysqli_fetch_assoc($result);
-    if($row)
-        $filter = $row["value"];
-    mysqli_free_result($result);
-
-    return $filter;
-}
+    $genre_element = new GenreElement("genre", _("Genre"), $data, $state_sql, $display_fields);
+    $pm_element = new ProjectFilterElement("username", _("Project Manager"), $data, $state_sql, $display_fields);
+    $pp_element = new ProjectFilterElement("checkedoutby", _("Post Processor"), $data, $state_sql, $display_fields);
+    $special_day_element = new SpecialDayElement("special_code", _("Special Days"), $data, $state_sql, $display_fields);
+    $difficulty_element = new DifficultyElement("difficulty", _("Difficulty"), $data, $state_sql, $display_fields);
+    $pri_lang_element = new LanguageElement("language", _("Language"), $data, $state_sql, $display_fields);
+    $elements = [
+        $genre_element,
+        $pm_element,
+        $pp_element,
+        $special_day_element,
+        $pri_lang_element,
+        $difficulty_element,
+    ];
+
+    echo "<form id='filter_form' method='post' action='#filter_form'>";
+    echo "<p>" . sprintf(_("Filter available %s books by:"), $filter_label) . "</p>";
+    echo "<table class='filter'>\n";
 
-function get_project_filter_display($username, $filter_type)
-{
-    $filter_display="";
+    // create the row for listboxes for Language, Genre, PM, Special Days, PPers (when enabled)
+    echo "<tr>\n";
 
-    $query = sprintf("SELECT value FROM user_filters WHERE username = '%s' AND filtertype = '%s_display'", mysqli_real_escape_string(DPDatabase::get_connection(), $username), mysqli_real_escape_string(DPDatabase::get_connection(), $filter_type));
-    $result = mysqli_query(DPDatabase::get_connection(), $query);
-    $row = mysqli_fetch_assoc($result);
-    if($row)
-        $filter_display = $row["value"];
-    mysqli_free_result($result);
+    $elements_on_first_line = [$genre_element, $pm_element, $pp_element, $special_day_element, $pri_lang_element];
 
-    return $filter_display;
-}
+    foreach($elements_on_first_line as $element)
+    {
+        $element->echo_html_control();
+    }
+    echo "</tr>\n";
+    echo "</table>\n";
 
-function get_current_project_filter_data($username, $filter_type)
-// Return a data array of the current project filter
-{
-    $filter=get_project_filter_sql($username, $filter_type);
-    return _populate_project_filter_data_from_filter($filter);
-}
+    $difficulty_element->echo_html_control();
 
-function display_current_project_filter_settings($username, $filter_type)
-// Print out the current project filter settings for a given user and filter type
-{
-    // if the username is blank (ie: the user isn't logged in) just return
-    // as there would be nothing to show them
-    if(empty($username))
-        return;
+    echo "<br><input type='submit' class='margin-half-em' name='reset' value='", attr_safe(_("Remove Filter")), "'>";
```

Comment:
We should either indent the difficulty a half-em too, or not indent these two a half-em to be consistent.

Reply:
removed class='margin-half-em'. Added spacing between the buttons